apps/store/src/features/widget/SelectProductPage.tsx

```diff
@@ -0,0 +1,141 @@
+}> = [
+  {
+    name: 'SE_WIDGET_APARTMENT_RENT',
+    title: 'Hyresrätt',
+    subtitle: 'För dig som hyr bostad',
+    pillow: {
+      src: 'https://a.storyblok.com/f/165473/832x832/fb3ddd4632/hedvig-pillows-rental.png',
+    },
+  },
+]
+
+type Props = {
+  flow: string
+  shopSessionId: string
+}
+
+export const SelectProductPage = (props: Props) => {
+  const [productName, setProductName] = useState<WidgetProductName | undefined>(undefined)
+
+  const router = useRouter()
+  const apolloClient = useApolloClient()
+  const handleSubmit: FormEventHandler<HTMLFormElement> = async (event) => {
+    event.preventDefault()
+    if (!productName) throw new Error('Missing product')
+
+    const priceIntentService = priceIntentServiceInitClientSide(apolloClient)
+    const priceIntent = await priceIntentService.create({
+      productName: productName,
+      priceTemplate: getPriceTemplate(productName),
+      shopSessionId: props.shopSessionId,
+    })
+
+    await router.push(
+      PageLink.widgetCalculatePrice({
+        flow: props.flow,
+        shopSessionId: props.shopSessionId,
+        priceIntentId: priceIntent.id,
+      }),
+    )
+  }
+
+  const handleValueChange = (value: string) => {
+    if (!isWidgetProductName(value)) throw new Error(`Invalid product: ${value}`)
+    setProductName(value)
+  }
+
+  return (
+    <Space y={4}>
+      <Header as="header">
+        <LogoArea>
+          <HedvigLogo />
+        </LogoArea>
+
+        <ProgressArea>
+          <ProgressIndicator.Root>
+            <ProgressIndicator.Step active={true}>Your info</ProgressIndicator.Step>
+            <ProgressIndicator.Step>Sign</ProgressIndicator.Step>
+            <ProgressIndicator.Step>Pay</ProgressIndicator.Step>
+          </ProgressIndicator.Root>
+        </ProgressArea>
+      </Header>
+
+      <GridLayout.Root>
+        <GridLayout.Content width="1/3" align="center">
+          <Space y={3.5}>
+            <Heading as="h1" variant="standard.24" align="center">
+              Select your insurance
+            </Heading>
+
+            <form onSubmit={handleSubmit}>
+              <Space y={0.5}>
+                <RadioOptionList.Root
+                  value={productName}
+                  onValueChange={handleValueChange}
+                  required={true}
+                >
+                  <Space y={0.5}>
+                    {PRODUCTS.map((item) => (
+                      <RadioOptionList.ProductOption
+                        key={item.name}
+                        value={item.name}
+                        title={item.title}
+                        subtitle={item.subtitle}
+                        pillow={item.pillow}
+                      />
+                    ))}
+                  </Space>
+                </RadioOptionList.Root>
+
+                <CustomButton type="submit" variant="primary" data-disabled={!productName}>
+                  Continue
+                </CustomButton>
```

Comment:
I saw that the idea is to keep the button clickable while it appears disabled when `productName` is `undefined`. Is that really necessary? I mean, by clicking on it while disabled makes the app break which is too harsh I think. 
If we could only prevent it to be clickable while disabled and use that state for applying styles, things would be simpler no? We can also add `aria-disabled` so assistive technologies can know about that. Do you think that would solve the A11y issues behind the addition of `data-disabled`?

```suggestion
                <CustomButton type="submit" variant="primary" disabled={"productName} aria-disabled={!productName}>
                  Continue
                </CustomButton>
```

Reply:
> by clicking on it while disabled makes the app break which is too harsh I think

Not sure what you mean here. If you click the button without selecting a product, we just show native form validation error.

> ...prevent it to be clickable while disabled and use that state for applying styles, things would be simpler no?

A disabled button is not clickable. However, in this case I want the button to remain clickable so assistive technologies can access it and support native error handling.

Aria labels is probably a good idea tho!

Reply:
Jaha, I wrote these because when testing this branch locally clicking on the button I as getting the following:

<img width="1027" alt="image" src="https://github.com/HedvigInsurance/racoon/assets/19200662/503c7b6b-0964-4ba4-b732-81858df307cf">

But doesn't happen anymore due the recent changes. Well, never mind then :)

Reply:
oh that's strange indeed -- then I totally understand your comment